apps/pp/serializers.py

```diff
@@ -23,6 +26,52 @@ class ResourceSerializer(ResourceIdSerializer, ResourceTypeSerializer):
     pass
 
 
+class LinkField(serializers.SerializerMethodField, serializers.URLField):
+    pass
+
+
+# Relation
+
+class ResourceLinksSerializer(serializers.Serializer):
+    self = LinkField()
+    self_link_url_name = None
+
+    def get_self(self, instance):
+        obj_id = getattr(instance, 'id', None) or instance
+        return reverse(self.self_link_url_name, args=(obj_id,))
+
+
+class RelationLinksSerializer(serializers.Serializer):
+    related = LinkField()
+    related_link_url_name = None
+
+    def get_related(self, instance):
+        serializer = self
+        related_link_url_name = getattr(serializer, 'related_link_url_name', None)
+        while related_link_url_name is None and serializer.parent is not None:
+            serializer = serializer.parent
+            related_link_url_name = getattr(serializer, 'related_link_url_name', None)
+        assert related_link_url_name is not None, \
```

Comment:
Oh, this just happened here :) nice

Reply:
This was more complicated case, use there is an assert. I agree though it should be in  get_self as well.